crates/lib/kajiya/src/world_renderer.rs

```diff
@@ -367,7 +367,11 @@ impl WorldRenderer {
             temporal_upscale_extent,
 
             debug_mode: RenderDebugMode::None,
-            debug_shading_mode: 0,
+            debug_shading_mode: if cfg!(feature = "ray-tracing") {
+                0
+            } else {
+                4
+            },
```

Comment:
This is also very messy. I'm not sure how nice defaults should be provided for both cases though.